### PR TITLE
fix for #1701

### PR DESCRIPTION
**Pull request by cristipogacean**
fix crash in case of invalid user authentication configuration

### CODE REVIEW

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/WriterGroupMessageSource.cs [hunk 1]
```diff
@@ -54,18 +54,18 @@ public class WriterGroupMessageTrigger : IMessageTrigger, IDisposable {
                 true : false;
 
         /// <inheritdoc/>
-        public OpcAuthenticationMode AuthenticationMode =>_subscriptions?.First()?.Subscription?.Connection.User != null ?
+        public OpcAuthenticationMode AuthenticationMode =>_subscriptions?.First()?.Subscription?.Connection?.User != null ?
```
Comment (karok2m) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/WriterGroupMessageSource.cs]: Should we use `_subscriptions?.First()?.Subscription?.Connection?.User?.Value` to keep the condition of `OpcAuthenticationMode` symmetric to that of `AuthenticationUsername` ?

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/WriterGroupMessageSource.cs [hunk 2]
```diff
@@ -54,18 +54,18 @@ public class WriterGroupMessageTrigger : IMessageTrigger, IDisposable {
...
-        public string AuthenticationUsername =>_subscriptions?.First()?.Subscription?.Connection.User != null ?
+        public string AuthenticationUsername =>_subscriptions?.First()?.Subscription?.Connection?.User?.Value != null ?
                 _serializer.Deserialize<cred>(_subscriptions?
                     .First()?
                     .Subscription?
-                    .Connection
-                    .User
-                    .Value
-                    .ToJson())
+                    .Connection?
```
Comment (karok2m) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/WriterGroupMessageSource.cs]: The condition above makes sure that `_subscriptions.First().Subscription.Connection.User.Value` is not `null` so we can remove the `?` checks up to `ToJson()` here.

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/WriterGroupMessageSource.cs (hunk 1 above)
Comment (cristipogacean) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/WriterGroupMessageSource.cs]: agreed

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/WriterGroupMessageSource.cs (hunk 2 above)
Comment (cristipogacean) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/WriterGroupMessageSource.cs]: agreed, thought originally better safe than sorry, but you're right.

**Review by karok2m — APPROVED** [review]

**Review by dacolgit — APPROVED** [review]

**Pull request merged**